mesa/space.py

```diff
@@ -808,11 +823,15 @@ def move_agent(self, agent: Agent, pos: FloatCoordinate) -> None:
             pos: Coordinate tuple to move the agent to.
         """
         pos = self.torus_adj(pos)
-        idx = self._agent_to_index[agent]
-        self._agent_points[idx, 0] = pos[0]
-        self._agent_points[idx, 1] = pos[1]
         agent.pos = pos
 
+        if self._agent_points is not None:
```

Comment:
Why is there no else: self._build_agent_cache() after this if?

Reply:
I see. Because the cache is only needed whenever `get_neighbors` is called.

Reply:
The cache only needs to be built directly before using it.
If you would add ```self._build_agent_cache()``` here, then moving an agent becomes the bottleneck (as each move includes recalculating the _agent_points``` etc.).